### PR TITLE
Datastore revamp 1: new indexing model & core datastructures

### LINKED ISSUE #445: re_datastore: gather thoughts on how streaming to/from disk will fit into all of this

### LINKED ISSUE #487: re_datastore: store component rows larger than a threshold behind an indirection
Let's say you have an arbitrarily large component (e.g. an `Image`, whose contents we have no control over).
As the contents of the `Image` gets larger and larger, the amount of rows we can fit in a single bucket gets smaller and smaller.
In the worst case, you get all of the overhead of bucketing, for none of the benefits.

Not only that, but it might also be the case that the contents of the `Image` are actually identical e.g. every other frame.
So not only we're wasting bucketing overhead, we're wasting space too.

In both situations, hashing the contents of the image if it gets beyond a certain threshold, and storing that hash instead of the actual contents, fixes the issue.
Of course, we need to store an extra map to resolve the hashes into actual data at query time.

TODO:
- [ ] implement
- [ ] test

### LINKED ISSUE #527: re_datastore: replace anyhow::Error usage with a thiserror derived Error type

### LINKED ISSUE #639: re_datastore: improve terminology for internal types and concepts
The implementation of range queries (#609) has put a _lot_ of stress on the terminology used within the store.
While it already wasn't that great before, it just completely broke down; making it really _really_ hard to understand what's going on in some cases.

Here are some notes I took to make things better / more understandable:
```rust
// TODO(cmc): We desperately need to work on the terminology here:
//
// - `TimeIndex` is a vector of `TimeInt`s.
//   It's the primary column and it's always dense.
//   It's used to search the datastore by time.
//
// - `ComponentIndex` (currently `SecondaryIndex`) is a vector of `ComponentRowNr`s.
//   It's the secondary column and is sparse.
//   It's used to refine the primary time-based search for a specific component.
//
// - `ComponentRowNr` (currently `RowIndex`) is a row offset into a component table.
//   It only makes sense when associated with a component name.
//   It is absolute.
//   It's used to fetch actual data from the datastore.
//
// - `IndexRowNr` is a row offset into an index bucket.
//   It only makes sense when associated with an entity path and a specific time.
//   It is relative per bucket.
//   It's used to tiebreak results with an identical time, should you need too.
```

### LINKED ISSUE #1746: `arrow2` erased refcounted clones performance issues
That one is _very_ bad, this is obviously something we do all over the place.

Background:
- Basically all arrow `Array`s are [`Buffer`s](https://docs.rs/arrow2/latest/arrow2/buffer/struct.Buffer.html) when you reach the actual physical type (which can be quite deep because all our components are compound types etc) 
- `Buffer` is literally an `Arc over some bytes, so cheap to clone

### PR DESCRIPTION
**Pull request by teh-cmc**
This PR kicks off the "datastore revamp" series of PRs.

This redesign results in much simpler internals, much less code, [major performance improvements](https://github.com/rerun-io/rerun/pull/1727#issuecomment-1494600232), for the same end result.  
Most notably, component tables are gone: all data is inlined directly within the indices, resulting in important performance gains on both the read and write path.

The `DataStore` is now effectively an infinitely large `DataTable`, designed to be capable of streaming `DataRow`s in and out as needed.  
Since it now uses the same datatypes as other layers in our stack (`DataTable`/`DataRow`/`DataCell`), it inherits from all their advantages:
- Tools: formatting, serialization, ...
- Transparent optimizations: memory is batched and made contiguous behind the scenes etc
- Centralized knowledge: all the pitfalls from `arrow2` and such in one single place

The end goal is to use these types across the entire stack: from the clients to the store to the disk to the UI to the renderer..!

Another important change with this PR in the introduction of control & data columns.  
Data columns are our usual component columns and are similar as before.  
Control columns are columns that drive the behavior of the storage systems and are always deserialized into native types upon reception:
- `row_id`
- `timepoint`
- `entity_path`
`EntityPath` is now just another column, making batching possible!
- `num_instances`
The number of instances is now explicitly set by the system logging the data: no more heuristic/guessing and no more nasty issues with splats!

Erased errors (`anyhow`) are also gone.

---

1. #1727 
2. #1735
3. #1739 
4. #1785 
5. #1791 
6. #1795
7. #1801

---

On top of #1733 
Part of #1619 

Closes #639 
Closes #527
Closes #487
Closes #445

Fixes #1746 by wrapping `DataCell`'s contents into an `Arc`.

Further improvements for later:
- #1696
- #1758
- #1759

---

- [x] Self-review
- [x] No bench regression
- [x] No test regression
- [x] No memory usage regression
- [x] Clocks 50k comparison
- [x] api_demo_rs looking good
- [x] api_demo_py looking good
- [x] TODO all the things that fit into #1619 
- [x] open issues for everything else

### CODE REVIEW

File: crates/re_log_types/src/data_table.rs [hunk 1]
```diff
@@ -34,13 +34,24 @@ pub enum DataTableError {
 
 pub type DataTableResult<T> = ::std::result::Result<T, DataTableError>;
 
+// TODO(cmc): The timepoint should be serialized as one column per timeline... that would be both
```
Comment (teh-cmc) [inline crates/re_log_types/src/data_table.rs:37]: open issue

**Comment by teh-cmc** [comment]
TODO:
- [ ] replace `smallvec` by `tinyvec` where possible (workaround for #1748)
- [ ] wrap `DataCell`'s body in an `Arc` to avoid costly virtual calls to `Array::clone` on the fast paths (workaround for #1746)
- [ ] compute size estimates for `DataCell`s out of the fast paths and cache the results (workaround for #1738)
- [ ] introduce a specialized fast path for serializing control columns backed by primitive types
- [ ] introduce a specialized fast path for serializing component columns backed by primitive types (`#[transparent]`)

**Comment by teh-cmc** [comment]
| Benchmark suite | Current: f07b8d81b74627a8f30dadacc045be486a937f1c | Previous: 29a13ea8fad4020c8c2c653f23e83e9a31a7beb1 | Ratio |
|-|-|-|-|
| `datastore/num_rows=1000/num_instances=1000/packed=false/insert/default` | `2441352` ns/iter (`± 29569`) | `10638674` ns/iter (`± 721417`) | `0.23` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/insert/bucketsz=0` | `3392186` ns/iter (`± 46761`) | `12800212` ns/iter (`± 452803`) | `0.27` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/insert/bucketsz=2` | `3414180` ns/iter (`± 43642`) | `12123770` ns/iter (`± 638739`) | `0.28` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/insert/bucketsz=32` | `2671023` ns/iter (`± 29313`) | `10600992` ns/iter (`± 454300`) | `0.25` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/insert/bucketsz=2048` | `2414547` ns/iter (`± 29792`) | `10847035` ns/iter (`± 710500`) | `0.22` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/insert/default` | `2304811` ns/iter (`± 17510`) | `10224985` ns/iter (`± 403984`) | `0.23` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/insert/bucketsz=0` | `3327011` ns/iter (`± 33483`) | `11999563` ns/iter (`± 580359`) | `0.28` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/insert/bucketsz=2` | `3368095` ns/iter (`± 47434`) | `11807113` ns/iter (`± 596072`) | `0.29` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/insert/bucketsz=32` | `2621769` ns/iter (`± 29175`) | `10482180` ns/iter (`± 413209`) | `0.25` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/insert/bucketsz=2048` | `2341808` ns/iter (`± 15129`) | `10184170` ns/iter (`± 455895`) | `0.23` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at/default` | `369` ns/iter (`± 5`) | `1828` ns/iter (`± 22`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at/bucketsz=0` | `373` ns/iter (`± 0`) | `1834` ns/iter (`± 12`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at/bucketsz=2` | `372` ns/iter (`± 0`) | `1846` ns/iter (`± 12`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at/bucketsz=32` | `371` ns/iter (`± 0`) | `1818` ns/iter (`± 8`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at/bucketsz=2048` | `368` ns/iter (`± 0`) | `1833` ns/iter (`± 6`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at/default` | `367` ns/iter (`± 4`) | `1821` ns/iter (`± 21`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at/bucketsz=0` | `371` ns/iter (`± 0`) | `1843` ns/iter (`± 15`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at/bucketsz=2` | `371` ns/iter (`± 1`) | `1824` ns/iter (`± 17`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at/bucketsz=32` | `370` ns/iter (`± 0`) | `1847` ns/iter (`± 5`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at/bucketsz=2048` | `367` ns/iter (`± 0`) | `1838` ns/iter (`± 20`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/primary/default` | `258` ns/iter (`± 0`) | `280` ns/iter (`± 1`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/secondaries/default` | `416` ns/iter (`± 0`) | `437` ns/iter (`± 1`) | `0.95` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/primary/bucketsz=0` | `259` ns/iter (`± 0`) | `279` ns/iter (`± 1`) | `0.93` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/secondaries/bucketsz=0` | `422` ns/iter (`± 2`) | `443` ns/iter (`± 1`) | `0.95` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/primary/bucketsz=2` | `258` ns/iter (`± 0`) | `280` ns/iter (`± 0`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/secondaries/bucketsz=2` | `422` ns/iter (`± 0`) | `444` ns/iter (`± 1`) | `0.95` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/primary/bucketsz=32` | `258` ns/iter (`± 0`) | `281` ns/iter (`± 0`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/secondaries/bucketsz=32` | `422` ns/iter (`± 1`) | `439` ns/iter (`± 3`) | `0.96` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/primary/bucketsz=2048` | `258` ns/iter (`± 0`) | `279` ns/iter (`± 0`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/latest_at_missing/secondaries/bucketsz=2048` | `417` ns/iter (`± 0`) | `435` ns/iter (`± 2`) | `0.96` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/primary/default` | `258` ns/iter (`± 0`) | `279` ns/iter (`± 0`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/secondaries/default` | `415` ns/iter (`± 0`) | `434` ns/iter (`± 1`) | `0.96` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/primary/bucketsz=0` | `257` ns/iter (`± 0`) | `279` ns/iter (`± 0`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/secondaries/bucketsz=0` | `421` ns/iter (`± 0`) | `447` ns/iter (`± 3`) | `0.94` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/primary/bucketsz=2` | `259` ns/iter (`± 0`) | `278` ns/iter (`± 1`) | `0.93` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/secondaries/bucketsz=2` | `427` ns/iter (`± 0`) | `449` ns/iter (`± 0`) | `0.95` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/primary/bucketsz=32` | `259` ns/iter (`± 0`) | `279` ns/iter (`± 0`) | `0.93` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/secondaries/bucketsz=32` | `421` ns/iter (`± 0`) | `439` ns/iter (`± 0`) | `0.96` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/primary/bucketsz=2048` | `258` ns/iter (`± 0`) | `280` ns/iter (`± 1`) | `0.92` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/latest_at_missing/secondaries/bucketsz=2048` | `415` ns/iter (`± 0`) | `434` ns/iter (`± 1`) | `0.96` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/range/default` | `2578085` ns/iter (`± 44515`) | `12174288` ns/iter (`± 597918`) | `0.21` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/range/bucketsz=0` | `310498` ns/iter (`± 370`) | `2181647` ns/iter (`± 27486`) | `0.14` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/range/bucketsz=2` | `314211` ns/iter (`± 384`) | `2186675` ns/iter (`± 14860`) | `0.14` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/range/bucketsz=32` | `117894` ns/iter (`± 261`) | `1936760` ns/iter (`± 12398`) | `0.06087176521613416` |
| `datastore/num_rows=1000/num_instances=1000/packed=false/range/bucketsz=2048` | `99129` ns/iter (`± 73`) | `1840328` ns/iter (`± 16573`) | `0.053864854525932336` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/range/default` | `2302562` ns/iter (`± 18265`) | `11722311` ns/iter (`± 742657`) | `0.20` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/range/bucketsz=0` | `296407` ns/iter (`± 588`) | `2130938` ns/iter (`± 71721`) | `0.14` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/range/bucketsz=2` | `295286` ns/iter (`± 353`) | `2154847` ns/iter (`± 17490`) | `0.14` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/range/bucketsz=32` | `116075` ns/iter (`± 96`) | `1884472` ns/iter (`± 28531`) | `0.06159550261293349` |
| `datastore/num_rows=1000/num_instances=1000/packed=true/range/bucketsz=2048` | `98042` ns/iter (`± 194`) | `1830844` ns/iter (`± 28354`) | `0.05355016593439965` |
| `mono_points_arrow/generate_message_bundles` | `39671234` ns/iter (`± 1215647`) | `46316431` ns/iter (`± 644918`) | `0.86` |
| `mono_points_arrow/generate_messages` | `162155543` ns/iter (`± 1082984`) | `183382126` ns/iter (`± 1497890`) | `0.88` |
| `mono_points_arrow/encode_log_msg` | `209348439` ns/iter (`± 858672`) | `224379598` ns/iter (`± 1031199`) | `0.93` |
| `mono_points_arrow/encode_total` | `417312752` ns/iter (`± 2316939`) | `452302399` ns/iter (`± 1815719`) | `0.92` |
| `mono_points_arrow/decode_log_msg` | `252579963` ns/iter (`± 919571`) | `270391565` ns/iter (`± 1237132`) | `0.93` |
| `mono_points_arrow/decode_message_bundles` | `87799046` ns/iter (`± 676298`) | `99694109` ns/iter (`± 1125172`) | `0.88` |
| `mono_points_arrow/decode_total` | `336089311` ns/iter (`± 1500402`) | `368423831` ns/iter (`± 2140655`) | `0.91` |
| `mono_points_arrow_batched/generate_message_bundles` | `37170364` ns/iter (`± 1013035`) | `36718603` ns/iter (`± 1354639`) | `1.01` |
| `mono_points_arrow_batched/generate_messages` | `10459978` ns/iter (`± 428324`) | `11364244` ns/iter (`± 674006`) | `0.92` |
| `mono_points_arrow_batched/encode_log_msg` | `1800850` ns/iter (`± 5612`) | `1802197` ns/iter (`± 4059`) | `1.00` |
| `mono_points_arrow_batched/encode_total` | `49586742` ns/iter (`± 1166382`) | `51982074` ns/iter (`± 4524698`) | `0.95` |
| `mono_points_arrow_batched/decode_log_msg` | `981712` ns/iter (`± 2414`) | `991987` ns/iter (`± 4324`) | `0.99` |
| `mono_points_arrow_batched/decode_message_bundles` | `20762627` ns/iter (`± 697730`) | `20640513` ns/iter (`± 1463771`) | `1.01` |
| `mono_points_arrow_batched/decode_total` | `22205122` ns/iter (`± 590393`) | `21075346` ns/iter (`± 1540303`) | `1.05` |
| `batch_points_arrow/generate_message_bundles` | `280692` ns/iter (`± 576`) | `284473` ns/iter (`± 1995`) | `0.99` |
| `batch_points_arrow/generate_messages` | `7523` ns/iter (`± 13`) | `7639` ns/iter (`± 13`) | `0.98` |
| `batch_points_arrow/encode_log_msg` | `382119` ns/iter (`± 869`) | `394503` ns/iter (`± 3369`) | `0.97` |
| `batch_points_arrow/encode_total` | `691869` ns/iter (`± 1957`) | `695338` ns/iter (`± 4884`) | `1.00` |
| `batch_points_arrow/decode_log_msg` | `338351` ns/iter (`± 633`) | `337876` ns/iter (`± 1156`) | `1.00` |
| `batch_points_arrow/decode_message_bundles` | `2984` ns/iter (`± 15`) | `2912` ns/iter (`± 13`) | `1.02` |
| `batch_points_arrow/decode_total` | `345590` ns/iter (`± 613`) | `347168` ns/iter (`± 3149`) | `1.00` |
| `arrow_mono_points/insert` | `1839188655` ns/iter (`± 17299011`) | `6958829824` ns/iter (`± 13837782`) | `0.26` |
| `arrow_mono_points/query` | `1597645` ns/iter (`± 7937`) | `1812590` ns/iter (`± 20101`) | `0.88` |
| `arrow_batch_points/insert` | `715672` ns/iter (`± 4139`) | `2970081` ns/iter (`± 36870`) | `0.24` |
| `arrow_batch_points/query` | `16845` ns/iter (`± 15`) | `16453` ns/iter (`± 50`) | `1.02` |
| `arrow_batch_vecs/insert` | `26575` ns/iter (`± 94`) | `42748` ns/iter (`± 203`) | `0.62` |
| `arrow_batch_vecs/query` | `534887` ns/iter (`± 487`) | `506242` ns/iter (`± 2723`) | `1.06` |

File: crates/re_arrow_store/src/store_write.rs [hunk 2]
```diff
@@ -259,191 +149,120 @@ impl DataStore {
...
-        //
-        // push new row indices to their associated secondary index
-        for (name, row_idx) in row_indices {
-            let index = self
-                .indices
-                .entry(*name)
-                .or_insert_with(|| vec![None; self.num_rows as usize]);
-            index.push(Some(*row_idx));
-        }
+        if let Some(cell) = self.cluster_cell_cache.get(&num_instances) {
```
Comment (emilk) [inline crates/re_arrow_store/src/store_write.rs:199]: One optimization we can do here in the future is to use slicing to reuse the same cache for many different sizes. That is: `generate_cluster_cell(42)` could return the `0..42` slice of a very long `UInt64Array`

File: crates/re_log_types/Cargo.toml [hunk 3]
```diff
@@ -76,7 +77,7 @@ ndarray.workspace = true
 nohash-hasher = "0.2"
 num-derive = "0.3"
 num-traits = "0.2"
-smallvec = "1.10"
+smallvec = { version = "1.10", features = ["union"] }
```
Comment (emilk) [inline crates/re_log_types/Cargo.toml:80]: we should add `smallvec` to the workspace `Cargo.toml`, including all opt-in features that are pure wins (and that only exist for backwards compatibility), including `union`

File: crates/re_log_types/src/data_table.rs [hunk 4]
```diff
@@ -347,20 +372,66 @@ impl DataTable {
...
+            col_num_instances,
+            columns,
+        } = self;
+
+        use itertools::Itertools as _;
+
+        let col_row_id = col_row_id.into_iter();
+        let col_timepoint = col_timepoint.into_iter();
+        let col_entity_path = col_entity_path.into_iter();
+        let col_num_instances = col_num_instances.into_iter();
```
Comment (emilk) [inline crates/re_log_types/src/data_table.rs:400]: I think these are superfluous - `izip` already calls `into_iter` on all arguments iirc

File: crates/re_log_types/src/data_table.rs [hunk 5]
```diff
@@ -347,20 +372,66 @@ impl DataTable {
...
+
+        let mut columns = columns
+            .into_values()
+            .map(|column| column.0.into_iter())
+            .collect_vec();
+        let rows = std::iter::from_fn(move || {
+            let mut next = Vec::with_capacity(columns.len());
```
Comment (emilk) [inline crates/re_log_types/src/data_table.rs:407]: this could end up wasting a lot of ram for sparse tables

**Review by emilk — APPROVED** [review]
woop-woop!

File: crates/re_arrow_store/src/store.rs [hunk 6]
```diff
@@ -1,194 +1,112 @@
...
-        //
-        // This has no noticeable impact on performance.
-        enable_compaction: false,
+        enable_typecheck: cfg!(debug_assertions),
     };
 }
 
 // ---
 
+pub type InsertIdVec = SmallVec<[u64; 4]>;
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_arrow_store/src/store.rs:65]: Feels unnecessary _for now_ imo

File: crates/re_arrow_store/src/store_read.rs [hunk 7]
```diff
@@ -1090,135 +897,192 @@ impl IndexBucketIndices {
...
+        // TODO(cmc): Cloning these is obviously not great and will need to be addressed at
+        // some point.
+        // But, really, it's not _that_ bad either: these are either integers or erased pointers,
+        // and e.g. with the default configuration there are only 1024 of them (times the number
+        // of components).
+        let row_id = self.col_row_id.clone();
+        let mut columns = self.columns.clone(); // shallow
+
+        let cells = (0..self.total_rows()).filter_map(move |row_nr| {
+            let mut cells = [(); N].map(|_| None);
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_arrow_store/src/store_read.rs:1059]: Nope that'd require `DataCell` to be `Copy`, shortcoming of the type system.. :disappointed:

File: crates/re_arrow_store/src/store_read.rs [hunk 8]
```diff
@@ -1090,135 +897,192 @@ impl IndexBucketIndices {
...
-        } else {
-            self.chunks.get(row_idx as usize).map(|chunk| {
-                chunk
-                    .as_any()
-                    .downcast_ref::<ListArray<i32>>()
-                    .unwrap()
-                    .value(0)
-            })
+    /// Returns an empty iterator if no data could be found for any reason.
+    pub fn range<const N: usize>(
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_arrow_store/src/store_read.rs:1034]: That's the private API though (the entire type is private to the crate); the public entrypoint has extensive doc already

File: crates/re_log_types/src/data_table.rs (hunk 5 above)
Comment (teh-cmc) [reply crates/re_log_types/src/data_table.rs:439]: this entire thing is actually pointless now that virtual calls are gone -> :wastebasket:

File: rerun_py/rerun_sdk/rerun/__init__.py [hunk 9]
```diff
@@ -484,19 +484,3 @@ def set_time_nanos(timeline: str, nanos: Optional[int]) -> None:
...
-    This is the same as calling `set_time_*` with `None` for all of the active timelines.
-
-    Used for all subsequent logging on the same thread,
-    until the next call to [`rerun.set_time_nanos`][] or [`rerun.set_time_seconds`][].
-    """
-
-    if not bindings.is_enabled():
-        return
-
-    bindings.reset_time()
```
Comment (jleibs) [inline rerun_py/rerun_sdk/rerun/__init__.py:502]: We accidentally reverted https://github.com/rerun-io/rerun/pull/1826 in this merge :-(

**Pull request merged**